data_steward/validation/main.py

```diff
@@ -730,15 +730,20 @@ def run_retraction_cron():
 
     # retract from bq
     dataset_ids = bq_utils.get_retraction_dataset_ids()
-    dataset_ids = dataset_ids.split() if dataset_ids else None
+    # If the env variable RETRACTION_DATASET_IDS does not exist, dataset_ids will be set to None
+    # This allows retraction to be performed on all datasets existing in the project
+    if dataset_ids is not None:
+        dataset_ids = dataset_ids.split()
+    logging.info('Dataset ids from env variable: %s' % dataset_ids)
     logging.info('Running retraction on BQ datasets')
     retract_data_bq.run_retraction(project_id, sandbox_dataset_id, pid_table_id, hpo_id, dataset_ids)
     logging.info('Completed retraction on BQ datasets')
 
     # retract from gcs
     folder = bq_utils.get_retraction_submission_folder()
-    if not folder:
-        folder = None
+    # If the env variable RETRACTION_SUBMISSION_FOLDER does not exist, folder will be set to None
+    # This allows retraction to be performed on all folders submitted by the site that exist in the archive
+    logging.info('Submission folder from env variable: %s of type %s' % (folder, type(folder)))
```

Comment:
nit:  This is an odd log message.  Literally comes out to either:

```
"Submission folder from env variable: alsdkfjaldskfj of type <class 'str'>"
"Submission folder from env variable: None of type <class 'NoneType'>"
```
You could drop the type and you would still have the useful part of the message.

Reply:
I was finally able to reproduce this behavior in an interpreter with the command line.  Maybe, we should redesign how you get your environment variables.  It seems that:
```
# in bash
$ export FAKE_VAR=

# in GAE app.yaml
FAKE_VAR:
``` 
produces an environment variable that is an empty string.  So it is technically not `None`.

One solution is to change your environment variable getters, (this may be something we need to investigate further for all environment variable getters).  Something like this would work:
```
def _get_env_var(var_name):
    """
    Get either a non-empty string environment variable or None.

    :param var_name:  name of the environment variable to look up
    :returns:  either a non-empty string or None
    """
    env_var = os.environ.get(var_name)
    env_var = env_var.strip()
    return env_var if env_var else None

def get_retraction_submission_folder():
    # submission folder from which to retract
    # if retraction needs to be performed on all submissions by a site, remove this env variable
    return _get_env_var('RETRACTION_SUBMISSION_FOLDER')
```

Reply:
Yes, locally we get an empty string, however in GAE we get the string 'None' for the same conditions rather than an empty string. One solution to this was to remove the environment variable from app.yaml. This returns None as expected both locally and in GAE.

Reply:
I added an additional check in the helper function.  Now, it should return `None` if the environment variable exists but is all spaces or if it is missing.  I feel like encapsulating this GAE specific logic in one location and setting an intuitive value is better than having this GAE logic spread out over multiple functions.

Reply:
I agree in centralizing it. However, we can do away with this check completely and add in our playbook a note saying 'no empty env variables allowed, remove if necessary'. I was exploring empty vars earlier but I believe the standard way to get `None` via os.environ.get` is to remove it completely. Also, the 'None' string that we get from GAE might be fixed/changed later on since it is unexpected/unexplained behavior, so I don't think we should rely on it.

Reply:
I'm hesitant about that because:  

1. This has caused you issues, costing debugging and development time.  
2.  It's nice to have a line in a developer guide, but I wouldn't rely on people to always read that every time they are going to deploy or develop new code.  
3.  This affects not just your retraction scripts, but any environment variable in app.yaml.
4.  This is not dependent on the environment variable returning `'None'`.  If the variable is missing or an empty string, it also returns `None`.
5.  This is a GAE specific quirk.  On any other system `os.environ.get` will default to `None`, not the string None.

Reply:
Thanks for the discussion, @lrwb-aou. I have used `all_datasets` and `all_folders` as the default values.